src/ResourceManager/Network/Commands.Network/Cortex/HubVnetConnection/NewAzureRmHubVirtualNetworkConnectionCommand.cs

```diff
@@ -0,0 +1,176 @@
+    using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+    using Microsoft.Azure.Management.Network;
+    using Microsoft.WindowsAzure.Commands.Common;
+    using MNM = Microsoft.Azure.Management.Network.Models;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using System.Linq;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+    [Cmdlet(VerbsCommon.New,
+        "AzureRmHubVirtualNetworkConnection",
+        DefaultParameterSetName = CortexParameterSetNames.ByVirtualHubName,
+        SupportsShouldProcess = true),
+        OutputType(typeof(PSHubVirtualNetworkConnection))]
+    public class NewHubVirtualNetworkConnectionCommand : HubVnetConnectionBaseCmdlet
+    {
+        [Alias("ResourceName", "HubVirtualNetworkConnectionName")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource name.")]
+        [ValidateNotNullOrEmpty]
+        public virtual string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubName,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public virtual string ResourceGroupName { get; set; }
+
+        [Alias("VirtualHubName", "ParentVirtualHubName")]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubName,
+            HelpMessage = "The parent resource name.")]
+        public string ParentResourceName { get; set; }
+
+        [Alias("VirtualHub", "ParentVirtualHub")]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubObject,
+            HelpMessage = "The parent resource.")]
+        public PSVirtualHub ParentResource { get; set; }
+
+        [Alias("VirtualHubId", "ParentVirtualHubId")]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubResourceId,
+            HelpMessage = "The parent resource id.")]
+        [ResourceIdCompleter("Microsoft.Network/virtualHubs")]
+        public string ParentResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The remote virtual network to which this hub virtual network connection is connected.")]
+        [ResourceGroupCompleter]
+        public PSVirtualNetwork RemoteVirtualNetwork { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
```

Comment:
You should have different parameter sets for RemoteVirtualNetwork and RemoteVirtualNetwork - all of your cmdlets should reflect the exact syntax we agreed upon in the review.

Reply:
We do allow both RemoteVirtualNetwork and RemoteVirtualNetworkId. I did not use parameter sets to reduce complexity of the handling.